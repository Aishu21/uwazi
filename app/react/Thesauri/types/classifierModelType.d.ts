/* eslint-disable */
/**AUTO-GENERATED. RUN yarn emit-types to update.*/

export interface ClassifierModelSchema {
  name: string;
  preferred?: string;
  config?: {
    bert: string;
    num_train?: number;
    num_test?: number;
    [k: string]: any | undefined;
  };
  completeness?: number;
  extraneous?: number;
  instances?: string[];
  topics: {
    [k: string]:
      | {
          name?: any;
          quality?: any;
          samples?: any;
          [k: string]: any | undefined;
        }
      | undefined;
  };
}
