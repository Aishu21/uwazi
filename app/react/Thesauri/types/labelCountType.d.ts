/* eslint-disable */
/**AUTO-GENERATED. RUN yarn emit-types to update.*/

export interface LabelCountSchema {
  totalRows: number;
  totalLabels: number;
  thesaurus: {
    propertyName: string;
    totalValues: {
      [k: string]: number | undefined;
    };
    [k: string]: unknown | undefined;
  };
}
