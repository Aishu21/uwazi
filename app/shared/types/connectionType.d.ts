/* eslint-disable */
/**AUTO-GENERATED. RUN yarn emit-types to update.*/

import { ObjectIdSchema } from 'shared/types/commonTypes';

import { EntitySchema } from 'shared/types/entityType';

export interface ConnectionSchema {
  _id?: string | ObjectId;
  hub?: string | ObjectId;
  template?: string | ObjectId;
  file?: string | ObjectId;
  entity?: unknown;
  entityData?: {
    _id?: string | ObjectId;
    sharedId?: string;
    language?: string;
    mongoLanguage?: string;
    title?: string;
    template?: string | ObjectId;
    published?: boolean;
    icon?: {
      _id?: string | null;
      label?: string;
      type?: string;
    };
    attachments?: {
      _id?: string | ObjectId;
      originalname?: string;
      filename?: string;
      mimetype?: string;
      timestamp?: number;
      size?: number;
      [k: string]: unknown | undefined;
    }[];
    creationDate?: number;
    user?: string | ObjectId;
    metadata?: {
      [k: string]:
        | {
            value:
              | null
              | string
              | number
              | {
                  label?: string | null;
                  url?: string | null;
                }
              | {
                  from?: number | null;
                  to?: number | null;
                }
              | {
                  label?: string;
                  lat: number;
                  lon: number;
                }
              | {
                  label?: string;
                  lat: number;
                  lon: number;
                }[];
            label?: string;
            suggestion_confidence?: number;
            suggestion_model?: string;
            provenance?: '' | 'BULK_ACCEPT';
            [k: string]: unknown | undefined;
          }[]
        | undefined;
    };
    suggestedMetadata?: {
      [k: string]:
        | {
            value:
              | null
              | string
              | number
              | {
                  label?: string | null;
                  url?: string | null;
                }
              | {
                  from?: number | null;
                  to?: number | null;
                }
              | {
                  label?: string;
                  lat: number;
                  lon: number;
                }
              | {
                  label?: string;
                  lat: number;
                  lon: number;
                }[];
            label?: string;
            suggestion_confidence?: number;
            suggestion_model?: string;
            provenance?: '' | 'BULK_ACCEPT';
            [k: string]: unknown | undefined;
          }[]
        | undefined;
    };
    [k: string]: unknown | undefined;
  };
  reference?: {
    text?: unknown;
    selectionRectangles?: {
      top?: number;
      left?: number;
      width?: number;
      height?: number;
      regionId?: string;
      [k: string]: unknown | undefined;
    }[];
    [k: string]: unknown | undefined;
  };
}
