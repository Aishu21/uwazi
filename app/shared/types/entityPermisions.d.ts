export interface MemberWithPermission {
  type: 'user' | 'group';
  _id: string;
  label: string;
  level?: 'read' | 'write' | 'mixed';
}
