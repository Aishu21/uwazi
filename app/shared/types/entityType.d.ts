/* eslint-disable */
/**AUTO-GENERATED. RUN yarn emit-types to update.*/

import { ObjectIdSchema, MetadataSchema, AttachmentSchema } from 'shared/types/commonTypes';

import { PermissionSchema } from 'shared/types/permissionType';

export interface EntitySchema {
  _id?: ObjectIdSchema;
  sharedId?: string;
  language?: string;
  mongoLanguage?: string;
  title?: string;
  template?: ObjectIdSchema;
  published?: boolean;
  icon?: {
    _id?: string | null;
    label?: string;
    type?: string;
  };
  attachments?: AttachmentSchema[];
  creationDate?: number;
  user?: ObjectIdSchema;
  metadata?: MetadataSchema;
  suggestedMetadata?: MetadataSchema;
  permissions?: PermissionSchema[];
  [k: string]: unknown | undefined;
}
