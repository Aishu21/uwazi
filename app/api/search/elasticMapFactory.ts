import { template } from 'api/relationships/specs/fixtures';
import { MetadataTemplate } from 'app/Templates/components/MetadataTemplate';
import { TemplateSchema } from 'shared/types/templateType';
import { isArrayLiteralExpression } from 'typescript';

const text = {
  type: 'text',
  index: true,
  omit_norms: true,
  analyzer: 'tokenizer',
  fields: {
    raw: { type: 'keyword' },
    sort: { type: 'text', fielddata: true, analyzer: 'string_sorter' },
  },
  term_vector: 'with_positions_offsets',
};

const noSorttext = {
  type: 'text',
  index: true,
  omit_norms: true,
  analyzer: 'tokenizer',
  term_vector: 'with_positions_offsets',
};

const noIndexText = {
  type: 'text',
  index: false,
};

const id = {
  type: 'keyword',
  index: true,
};

const date = { type: 'date', doc_values: true };

const geolocation = { type: 'object' };
const nested = { type: 'nested' };
const number = {
  type: 'double',
  doc_values: true,
  fields: {
    raw: { type: 'double', index: false },
    sort: { type: 'double' },
  },
};

const noSortNumber = {
  type: 'double',
  doc_values: true,
};

const textType = () => {
  return {
    properties: {
      label: text,
      value: text,
    },
  };
};

const dateType = () => {
  return {
    properties: {
      label: text,
      value: date,
    },
  };
};

const daterangeType = () => {
  return {
    properties: {
      label: text,
      value: {
        properties: {
          from: date,
          to: date,
        },
      },
    },
  };
};

const geolocationType = () => {
  return {
    properties: {
      value: {
        properties: {
          label: text,
          lat: noSortNumber,
          lon: noSortNumber,
        },
      },
    },
  };
};

const imageType = () => {
  return {
    properties: {
      value: noIndexText,
    },
  };
};

const linkType = () => {
  return {
    properties: {
      value: {
        properties: {
          label: text,
          url: noIndexText,
        },
      },
    },
  };
};

const markdownType = () => {
  return {
    properties: {
      value: noSorttext,
    },
  };
};

const selectType = () => {
  return {
    properties: {
      label: text,
      value: id,
    },
  };
};

const noMapping = () => {
  return;
};

const numericType = () => {
  return {
    properties: {
      value: number,
    },
  };
};

const relationshipType = () => {
  return {
    properties: {
      label: text,
      value: id,
      type: noIndexText,
    },
  };
};

const mappings = {
  text: textType,
  date: dateType,
  daterange: daterangeType,
  geolocation: geolocationType,
  image: imageType,
  link: linkType,
  markdown: markdownType,
  media: imageType,
  multidate: dateType,
  multidaterange: daterangeType,
  multiselect: selectType,
  nested: noMapping,
  numeric: numericType,
  preview: noMapping,
  relationship: relationshipType,
  select: selectType,
};

export default {
  mapping: (template: TemplateSchema) => {
    return template.properties?.reduce((map: any, property) => {
      if (!property.name || !property.type) {
        return map;
      }

      map[property.name] = {
        match: property.name,
        match_mapping_type: 'string',
        mapping: mappings[property.type](),
      };

      return map;
    }, {});
  },

  aliasses: (templates: TemplateSchema[]) => {
    const baseAliasObject = {
      properties: {
        metadata: {
          properties: {},
        },
      },
    };

    return templates.reduce((baseAlias: any, template: TemplateSchema) => {
      return template.properties?.reduce((aliases: any, property) => {
        if (!property.name || !property.type) {
          return aliases;
        }

        aliases.properties.metadata.properties[`${property.name}_${property.type}.value`] = {
          type: 'alias',
          path: `metadata.${property.name}.value`,
        };

        aliases.properties.metadata.properties[`${property.name}_${property.type}.label`] = {
          type: 'alias',
          path: `metadata.${property.name}.label`,
        };

        return aliases;
      }, baseAlias);
    }, baseAliasObject);
  },

  ingest: (templates: TemplateSchema[]) => {
    const basePipelineObject = {
      description: 'rename pipeline',
      processors: [],
    };
    return templates.reduce((basePipeline, template) => {
      return template.properties?.reduce((pipeline: any, property) => {
        if (!property.name || !property.type) {
          return pipeline;
        }

        pipeline.processors.push({
          rename: {
            field: property.name,
            target_field: `${property.name}_${property.type}`,
          },
        });

        return pipeline;
      }, basePipeline);
    }, basePipelineObject);
  },
};
