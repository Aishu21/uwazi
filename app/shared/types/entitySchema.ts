import Ajv from 'ajv';
import templatesModel from 'api/templates/templatesModel';
import { isUndefined, isNull } from 'util';
import { objectIdSchema, metadataSchema } from 'shared/types/commonSchemas';
import { wrapValidator, ensure } from 'shared/tsUtils';
import { validators, customErrorMessages } from 'api/entities/metadataValidators.js';
import { EntitySchema } from './entityType';
import { PropertySchema } from './commonTypes';
import { TemplateSchema } from './templateType';

export const emitSchemaTypes = true;

const ajv = Ajv({ allErrors: true });

const hasValue = (value: any) => !isUndefined(value) && !isNull(value);

const validateMetadataField = (property: PropertySchema, entity: EntitySchema) => {
  const value = entity.metadata && entity.metadata[ensure<string>(property.name)];

  if (!validators.validateRequiredProperty(property, value)) {
    throw new Error(customErrorMessages.required);
  }

  //@ts-ignore
  if (hasValue(value) && validators[property.type] && !validators[property.type](value)) {
    //@ts-ignore
    throw new Error(customErrorMessages[property.type]);
  }

  if (hasValue(value) && !validators.validateLuceneBytesLimit(value)) {
    throw new Error(customErrorMessages.length_exceeded);
  }
};

ajv.addKeyword('metadataMatchesTemplateProperties', {
  async: true,
  errors: true,
  type: 'object',
  async validate(fields: any, entity: EntitySchema) {
    if (!entity.template) {
      return true;
    }
    const [template = {} as TemplateSchema] = await templatesModel.get({ _id: entity.template });
    const errors: Ajv.ErrorObject[] = (template.properties || []).reduce<Ajv.ErrorObject[]>(
      (err: Ajv.ErrorObject[], property) => {
        try {
          validateMetadataField(property, entity);
          return err;
        } catch (e) {
          err.push({
            keyword: 'metadataMatchesTemplateProperties',
            schemaPath: '',
            params: { keyword: 'metadataMatchesTemplateProperties', fields },
            message: e.message,
            dataPath: `.metadata['${property.name}']`,
          });
          return err;
        }
      },
      []
    );

    if (errors.length) {
      throw new Ajv.ValidationError(errors);
    }

    return true;
  },
});

ajv.addKeyword('validateTemplateExists', {
  async: true,
  errors: true,
  type: 'object',
  async validate(fields: any, entity: EntitySchema) {
    if (!entity.template) {
      return true;
    }
    const [template] = await templatesModel.get({ _id: entity.template });
    if (!template) {
      throw new Ajv.ValidationError([
        {
          keyword: 'metadataMatchesTemplateProperties',
          schemaPath: '',
          params: { keyword: 'metadataMatchesTemplateProperties', fields },
          message: 'template does not exist',
          dataPath: '.template',
        },
      ]);
    }
    return true;
  },
});

ajv.addKeyword('stringMeetsLuceneMaxLimit', {
  errors: false,
  type: 'string',
  validate(_schema: any, data: EntitySchema) {
    return validators.validateLuceneBytesLimit(data);
  },
});

export const entitySchema = {
  $schema: 'http://json-schema.org/schema#',
  $async: true,
  type: 'object',
  metadataMatchesTemplateProperties: true,
  validateTemplateExists: true,
  definitions: {
    objectIdSchema,
    metadataSchema,
  },
  properties: {
    _id: objectIdSchema,
    sharedId: { type: 'string', minLength: 1 },
    language: { type: 'string', minLength: 1 },
    mongoLanguage: { type: 'string' },
    title: { type: 'string', minLength: 1, stringMeetsLuceneMaxLimit: true },
    template: objectIdSchema,
    published: { type: 'boolean' },
    icon: {
      type: 'object',
      additionalProperties: false,
      properties: {
        _id: { anyOf: [{ type: 'string' }, { type: 'null' }] },
        label: { type: 'string' },
        type: { type: 'string' },
      },
    },
    attachments: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          originalname: { type: 'string' },
          filename: { type: 'string' },
          mimetype: { type: 'string' },
          timestamp: { type: 'number' },
          size: { type: 'number' },
        },
      },
    },
    creationDate: { type: 'number' },
    user: objectIdSchema,
    metadata: metadataSchema,
    suggestedMetadata: metadataSchema,
  },
};

export const validateEntity = wrapValidator(ajv.compile(entitySchema));
