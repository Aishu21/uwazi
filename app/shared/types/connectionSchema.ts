import { objectIdSchema } from 'shared/types/commonSchemas';
import { entitySchema } from 'shared/types/entitySchema';

export const emitSchemaTypes = true;

export const connectionSchema = {
  definitions: { objectIdSchema },
  type: 'object',
  additionalProperties: false,
  properties: {
    _id: objectIdSchema,
    hub: objectIdSchema,
    template: objectIdSchema,
    file: objectIdSchema,
    entity: 'string',
    entityData: entitySchema,
    reference: {
      type: 'object',
      properties: {
        text: 'string',
        selectionRectangles: {
          type: 'array',
          items: {
            type: 'object',
            properties: {
              top: { type: 'number' },
              left: { type: 'number' },
              width: { type: 'number' },
              height: { type: 'number' },
              regionId: { type: 'string' },
            },
          },
        },
      },
    },
  },
};
