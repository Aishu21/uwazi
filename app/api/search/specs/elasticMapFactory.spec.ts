import { TemplateSchema } from 'shared/types/templateType';
import elasticMapFactory from '../elasticMapFactory';

describe('elasticMapFactory', () => {
  let templates: TemplateSchema[];
  let template: TemplateSchema;
  beforeEach(() => {
    template = {
      name: 'string',
      properties: [
        { label: 'Name', name: 'name', type: 'text' },
        { label: 'Date of birth', name: 'date_of_birth', type: 'date' },
        { label: 'Country', name: 'country', type: 'select' },
        { label: 'Years in prison', name: 'years_in_prison', type: 'daterange' },
        { label: 'Prison location', name: 'prisonLocation', type: 'geolocation' },
        { label: 'Preview', name: 'preview', type: 'preview' },
        { label: 'Photo', name: 'photo', type: 'image' },
        { label: 'External profile', name: 'external_profile', type: 'link' },
        { label: 'Number of incidents', name: 'number_of_incidents', type: 'numeric' },
      ],
    };
    templates = [template];
  });

  describe('mapping()', () => {
    it('should return a mapping based on a template', () => {
      const mapping = elasticMapFactory.mapping(template);
      expect(mapping).toMatchSnapshot();
    });
  });

  describe('aliases()', () => {
    it('should return the alias config for the templates', () => {
      const aliasesDefinition = elasticMapFactory.aliasses(templates);
      expect(aliasesDefinition.properties.metadata.properties['name_text.value']).toEqual({
        path: 'metadata.name.value',
        type: 'alias',
      });
      expect(aliasesDefinition).toMatchSnapshot();
    });
  });

  describe('ingest()', () => {
    it('should return the alias config for the templates', () => {
      const ingestpipeline = elasticMapFactory.ingest(templates);
      expect(ingestpipeline.description).toBe('rename pipeline');

      expect(ingestpipeline.processors[0]).toEqual({
        rename: { field: 'name', target_field: 'name_text' },
      });
      expect(ingestpipeline.processors[1]).toEqual({
        rename: {
          field: 'date_of_birth',
          target_field: 'date_of_birth_date',
        },
      });

      expect(ingestpipeline).toMatchSnapshot();
    });
  });
});
