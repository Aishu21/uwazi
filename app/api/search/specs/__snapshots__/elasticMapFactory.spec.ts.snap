// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elasticMapFactory aliases() should return the alias config for the templates 1`] = `
Object {
  "properties": Object {
    "metadata": Object {
      "properties": Object {
        "country_select.label": Object {
          "path": "metadata.country.label",
          "type": "alias",
        },
        "country_select.value": Object {
          "path": "metadata.country.value",
          "type": "alias",
        },
        "date_of_birth_date.label": Object {
          "path": "metadata.date_of_birth.label",
          "type": "alias",
        },
        "date_of_birth_date.value": Object {
          "path": "metadata.date_of_birth.value",
          "type": "alias",
        },
        "external_profile_link.label": Object {
          "path": "metadata.external_profile.label",
          "type": "alias",
        },
        "external_profile_link.value": Object {
          "path": "metadata.external_profile.value",
          "type": "alias",
        },
        "name_text.label": Object {
          "path": "metadata.name.label",
          "type": "alias",
        },
        "name_text.value": Object {
          "path": "metadata.name.value",
          "type": "alias",
        },
        "number_of_incidents_numeric.label": Object {
          "path": "metadata.number_of_incidents.label",
          "type": "alias",
        },
        "number_of_incidents_numeric.value": Object {
          "path": "metadata.number_of_incidents.value",
          "type": "alias",
        },
        "photo_image.label": Object {
          "path": "metadata.photo.label",
          "type": "alias",
        },
        "photo_image.value": Object {
          "path": "metadata.photo.value",
          "type": "alias",
        },
        "preview_preview.label": Object {
          "path": "metadata.preview.label",
          "type": "alias",
        },
        "preview_preview.value": Object {
          "path": "metadata.preview.value",
          "type": "alias",
        },
        "prisonLocation_geolocation.label": Object {
          "path": "metadata.prisonLocation.label",
          "type": "alias",
        },
        "prisonLocation_geolocation.value": Object {
          "path": "metadata.prisonLocation.value",
          "type": "alias",
        },
        "years_in_prison_daterange.label": Object {
          "path": "metadata.years_in_prison.label",
          "type": "alias",
        },
        "years_in_prison_daterange.value": Object {
          "path": "metadata.years_in_prison.value",
          "type": "alias",
        },
      },
    },
  },
}
`;

exports[`elasticMapFactory ingest() should return the alias config for the templates 1`] = `
Object {
  "description": "rename pipeline",
  "processors": Array [
    Object {
      "rename": Object {
        "field": "name",
        "target_field": "name_text",
      },
    },
    Object {
      "rename": Object {
        "field": "date_of_birth",
        "target_field": "date_of_birth_date",
      },
    },
    Object {
      "rename": Object {
        "field": "country",
        "target_field": "country_select",
      },
    },
    Object {
      "rename": Object {
        "field": "years_in_prison",
        "target_field": "years_in_prison_daterange",
      },
    },
    Object {
      "rename": Object {
        "field": "prisonLocation",
        "target_field": "prisonLocation_geolocation",
      },
    },
    Object {
      "rename": Object {
        "field": "preview",
        "target_field": "preview_preview",
      },
    },
    Object {
      "rename": Object {
        "field": "photo",
        "target_field": "photo_image",
      },
    },
    Object {
      "rename": Object {
        "field": "external_profile",
        "target_field": "external_profile_link",
      },
    },
    Object {
      "rename": Object {
        "field": "number_of_incidents",
        "target_field": "number_of_incidents_numeric",
      },
    },
  ],
}
`;

exports[`elasticMapFactory mapping() should return a mapping based on a template 1`] = `
Object {
  "country": Object {
    "mapping": Object {
      "properties": Object {
        "label": Object {
          "analyzer": "tokenizer",
          "fields": Object {
            "raw": Object {
              "type": "keyword",
            },
            "sort": Object {
              "analyzer": "string_sorter",
              "fielddata": true,
              "type": "text",
            },
          },
          "index": true,
          "omit_norms": true,
          "term_vector": "with_positions_offsets",
          "type": "text",
        },
        "value": Object {
          "index": true,
          "type": "keyword",
        },
      },
    },
    "match": "country",
    "match_mapping_type": "string",
  },
  "date_of_birth": Object {
    "mapping": Object {
      "properties": Object {
        "label": Object {
          "analyzer": "tokenizer",
          "fields": Object {
            "raw": Object {
              "type": "keyword",
            },
            "sort": Object {
              "analyzer": "string_sorter",
              "fielddata": true,
              "type": "text",
            },
          },
          "index": true,
          "omit_norms": true,
          "term_vector": "with_positions_offsets",
          "type": "text",
        },
        "value": Object {
          "doc_values": true,
          "type": "date",
        },
      },
    },
    "match": "date_of_birth",
    "match_mapping_type": "string",
  },
  "external_profile": Object {
    "mapping": Object {
      "properties": Object {
        "value": Object {
          "properties": Object {
            "label": Object {
              "analyzer": "tokenizer",
              "fields": Object {
                "raw": Object {
                  "type": "keyword",
                },
                "sort": Object {
                  "analyzer": "string_sorter",
                  "fielddata": true,
                  "type": "text",
                },
              },
              "index": true,
              "omit_norms": true,
              "term_vector": "with_positions_offsets",
              "type": "text",
            },
            "url": Object {
              "index": false,
              "type": "text",
            },
          },
        },
      },
    },
    "match": "external_profile",
    "match_mapping_type": "string",
  },
  "name": Object {
    "mapping": Object {
      "properties": Object {
        "label": Object {
          "analyzer": "tokenizer",
          "fields": Object {
            "raw": Object {
              "type": "keyword",
            },
            "sort": Object {
              "analyzer": "string_sorter",
              "fielddata": true,
              "type": "text",
            },
          },
          "index": true,
          "omit_norms": true,
          "term_vector": "with_positions_offsets",
          "type": "text",
        },
        "value": Object {
          "analyzer": "tokenizer",
          "fields": Object {
            "raw": Object {
              "type": "keyword",
            },
            "sort": Object {
              "analyzer": "string_sorter",
              "fielddata": true,
              "type": "text",
            },
          },
          "index": true,
          "omit_norms": true,
          "term_vector": "with_positions_offsets",
          "type": "text",
        },
      },
    },
    "match": "name",
    "match_mapping_type": "string",
  },
  "number_of_incidents": Object {
    "mapping": Object {
      "properties": Object {
        "value": Object {
          "doc_values": true,
          "fields": Object {
            "raw": Object {
              "index": false,
              "type": "double",
            },
            "sort": Object {
              "type": "double",
            },
          },
          "type": "double",
        },
      },
    },
    "match": "number_of_incidents",
    "match_mapping_type": "string",
  },
  "photo": Object {
    "mapping": Object {
      "properties": Object {
        "value": Object {
          "index": false,
          "type": "text",
        },
      },
    },
    "match": "photo",
    "match_mapping_type": "string",
  },
  "preview": Object {
    "mapping": undefined,
    "match": "preview",
    "match_mapping_type": "string",
  },
  "prisonLocation": Object {
    "mapping": Object {
      "properties": Object {
        "value": Object {
          "properties": Object {
            "label": Object {
              "analyzer": "tokenizer",
              "fields": Object {
                "raw": Object {
                  "type": "keyword",
                },
                "sort": Object {
                  "analyzer": "string_sorter",
                  "fielddata": true,
                  "type": "text",
                },
              },
              "index": true,
              "omit_norms": true,
              "term_vector": "with_positions_offsets",
              "type": "text",
            },
            "lat": Object {
              "doc_values": true,
              "type": "double",
            },
            "lon": Object {
              "doc_values": true,
              "type": "double",
            },
          },
        },
      },
    },
    "match": "prisonLocation",
    "match_mapping_type": "string",
  },
  "years_in_prison": Object {
    "mapping": Object {
      "properties": Object {
        "label": Object {
          "analyzer": "tokenizer",
          "fields": Object {
            "raw": Object {
              "type": "keyword",
            },
            "sort": Object {
              "analyzer": "string_sorter",
              "fielddata": true,
              "type": "text",
            },
          },
          "index": true,
          "omit_norms": true,
          "term_vector": "with_positions_offsets",
          "type": "text",
        },
        "value": Object {
          "properties": Object {
            "from": Object {
              "doc_values": true,
              "type": "date",
            },
            "to": Object {
              "doc_values": true,
              "type": "date",
            },
          },
        },
      },
    },
    "match": "years_in_prison",
    "match_mapping_type": "string",
  },
}
`;
